src/utils/router_state.rs

```diff
@@ -260,15 +262,116 @@ pub fn is_vertiport_available(
             }
         })
         .count();
+    let res = if num_vertipads > 1 {
+        let vehicles_at_vertiport =
+            get_all_vehicles_scheduled_for_vertiport(&vertiport_id, date_to, existing_flight_plans);
+        (
+            vehicles_at_vertiport.len() < num_vertipads,
+            vehicles_at_vertiport,
+        )
+    } else {
+        (conflicting_flight_plans_count == 0, vec![])
+    };
     debug!(
         "Checking {} is departure: {}, is available for {} - {}? {}",
-        vertiport_id,
-        is_departure_vertiport,
-        date_from,
-        date_to,
-        conflicting_flight_plans_count == 0,
+        vertiport_id, is_departure_vertiport, date_from, date_to, res.0,
+    );
+    res
+}
+
+///Finds all vehicles which are parked at or in flight to the vertiport at specific timestamp
+/// Returns vector of tuples of (vehicle_id, minutes_to_arrival) where minutes_to_arrival is 0 if vehicle is parked at the vertiport
+/// and up to 10 minutes if vehicle is landing
+pub fn get_all_vehicles_scheduled_for_vertiport(
+    vertiport_id: &str,
+    timestamp: DateTime<Tz>,
+    existing_flight_plans: &[FlightPlan],
+) -> Vec<(String, i64)> {
+    let mut vehicles_plans_sorted: HashMap<String, Vec<FlightPlan>> = HashMap::new();
+    existing_flight_plans
+        .iter()
+        .filter(|flight_plan| {
+            flight_plan
+                .data
+                .as_ref()
+                .unwrap()
+                .destination_vertiport_id
+                .as_ref()
+                .unwrap()
+                == vertiport_id
+                && flight_plan
+                    .data
+                    .as_ref()
+                    .unwrap()
+                    .scheduled_arrival
+                    .as_ref()
+                    .unwrap()
+                    .seconds // arrival time needs to be less than 2x time needed - to allow landing and and then take off again)
+                    < timestamp.timestamp() + LANDING_AND_UNLOADING_TIME_MIN as i64 * 60
+        })
+        .for_each(|flight_plan| {
+            if vehicles_plans_sorted.contains_key(&flight_plan.data.as_ref().unwrap().vehicle_id) {
+                vehicles_plans_sorted
+                    .get_mut(&flight_plan.data.as_ref().unwrap().vehicle_id)
+                    .unwrap()
+                    .push(flight_plan.clone());
+            } else {
+                vehicles_plans_sorted.insert(
+                    flight_plan.data.as_ref().unwrap().vehicle_id.clone(),
+                    vec![flight_plan.clone()],
+                );
+            }
+        });
```

Comment:
Can be refactored using [`or_insert_with`](https://doc.rust-lang.org/stable/std/collections/hash_map/enum.Entry.html#method.or_insert_with)